src/main/scala/is/hail/methods/LinearMixedRegression.scala

```diff
@@ -151,9 +151,9 @@ object LinearMixedRegression {
 
       vds2.mapAnnotations { case (v, va, gs) =>
         val SparseGtVectorAndStats(x0, isConstant, af, nHomRef, nHet, nHomVar, nMissing) =
-          gs.iterator.zipWithIndex
-            .filter { case (g, i) => sampleMaskBc.value(i)}
-            .foldLeft(new SparseGtBuilder()){ case (b, (g, i)) => b.merge(g) }
+          gs.hardCallIterator.zipWithIndex
+            .filter { case (gt, i) => sampleMaskBc.value(i)}
+            .foldLeft(new SparseGtBuilder()){ case (b, (gt, i)) => b.merge(gt) }
```

Comment:
No, no, no.  This boxes all over the place.  You can't use the Iterator[Int] interfaces.  I allowed that for compatibility but I now think it was a mistake.  Remove `extends Iterator[Int]` from IntIterator and try again.